Gemfile.lock

```diff
@@ -0,0 +1,179 @@
+GIT
```

Comment:
I am aware of this discussion and I don't necessarily agree. It will depend very much on your workflow and whether or not you regularly delete your Gemfile.lock. If not, one can detect a potential gem conflict only on the first checkout and 'bundle install'. 

Personally, I'd rather ensure that we have a consistent dev environment and do dedicated bundle upgrades/checks as prep work for a release. 

Alternatively, one sets up a dedicated CI jobs which only job it is to test each commit against a fresh 'bundle install' (aka deleting the lock file). If it fails, someone can have a look and fix the issue. On the plus side, each developer checking out the repository and trying to build it will have a consistent experience. This might give you the best of both worlds.